lib/ansible/executor/task_executor.py

```diff
@@ -605,8 +605,10 @@ def _execute(self, variables=None):
 
         if module_context is not None:
             module_defaults_fqcn = module_context.resolved_fqcn
-        else:
+        elif self._task.resolved_action is not None:
             module_defaults_fqcn = self._task.resolved_action
+        else:
+            module_defaults_fqcn = self._handler.ansible_name
```

Comment:
This fixes action plugins, but it looks like modules have the same issue.

```yaml
- hosts: localhost
  gather_facts: false
  module_defaults:
    ansible.builtin.ping:
      data: MODULE_DEFAULT
  tasks:
    - ping:
    - action: ping
    - local_action:
        module: ping
```

Reply:
need another case for 'normal.py'

Reply:
Using the module name should be right for normal.py too, since it doesn't have any specific options?

Reply:
fixed by remove the 'nonization' of cases when there is an action plugin, should only be 'none' if no module/action was matched